runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/blocktransfer/BlockInputStream.java

```diff
@@ -136,6 +137,13 @@ void startDecodingThreadIfNeeded() {
     executorService.submit(() -> {
       try {
         final long startTime = System.currentTimeMillis();
+        if (byteBufInputStream.isEnded()) {
```

Comment:
Can you kindly add in-line comments above this line to clarify the reason this block is needed?